### ISSUE TITLE
Memory leak in saving and loading a keras model containing CategoricalEncoding and Lookup layers

### ISSUE THREAD

**Issue by AdrianFuchsData** [tensorflow/tensorflow#53435]
### System information

-   **Have I written custom code (as opposed to using a stock example script
    provided in TensorFlow)**: Yes
-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: CentOS Linux release 7.9.2009 (Core)
-   **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue
    happens on a mobile device**: not applicable
-   **TensorFlow installed from (source or binary)**: binary
-   **TensorFlow version (use command below)**: 2.6.0 (v2.6.0-rc2-32-g919f693420e), 2.7.0 (v2.7.0-rc1-69-gc256c071bb2)
-   **Python version**: 3.7.11, 3.8.12, 3.9.6
-   **Bazel version (if compiling from source)**: not applicable
-   **GCC/Compiler version (if compiling from source)**: not applicable
-   **CUDA/cuDNN version**: no GPU available
-   **GPU model and memory**: no GPU available
-   **Exact command to reproduce**: See code below

### Describe the problem

To solve a binary classification problem, I have a keras model that processes categorical input (as as well as numeric input).  
I need to save (`model.save`) and load (`tf.keras.models.load_model`) the model multiple times (performig training of the model inbetween).  

I expect that the model consumes constant disk space and constant RAM everytime I load the model since the architecture does not change (only the parameter values change).
This does not happen when the model contains an `IntegerLookup` layer followed by a `CategoryEncoding` layer. 

The issue can be reproduced without training the model at all.  
Here is a minimal code example that creates a model and saves it to disk:

    import tensorflow as tf
    import numpy as np

    input_layer = tf.keras.Input(shape=(1,), dtype="int32") 
    index = tf.keras.layers.IntegerLookup(max_values=2)
    index.adapt(np.array(range(2)))
    encoder = tf.keras.layers.CategoryEncoding(max_tokens=index.vocab_size())
    encoded_layer = encoder(index(input_layer))
    output_layer = tf.keras.layers.Dense(2, activation=tf.keras.activations.softmax)(encoded_layer)
    model = tf.keras.Model(input_layer, output_layer)
    model.compile(optimizer="adam", loss="sparse_categorical_crossentropy")
    model.save("model")
    
Every time I execute

    model = tf.keras.models.load_model("model")
    model.save("model")

the space that the model consumes on disk increases by approx. 8 kB.
The even worse: When I load the model, the RAM useage increases by approx. 9 MB in each iteration.
So after 100 iterations, the model needs approx. 1 MB on disk and 950 MB RAM (which *is* problematic).

This also happens if I start a new python process in each iteration.

In my application, the memory consumption grows even faster because the model has several input layers and also several inner layers.
This makes the model unusable after some iterations because I cannot load it anymore.

Additionaly, of course, the load and save cycles are getting slower with each repetition.

So far, I could reproduce this issue on tensorflow versions 2.6 and 2.7 running on python 3.7, 3.8 or 3.9. The behavior is identical.

**Comment by sushreebarsa** [comment]
@AdrianFuchsData 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thanks!

**Comment by AdrianFuchsData** [comment]
Alright, let's discuss the issue here: https://github.com/keras-team/keras/issues/15807
